connectors/sources/google_drive.py

```diff
@@ -0,0 +1,638 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import asyncio
+import json
+import os
+from functools import cached_property, partial
+
+import aiofiles
+from aiofiles.os import remove
+from aiofiles.tempfile import NamedTemporaryFile
+from aiogoogle import Aiogoogle, HTTPError
+from aiogoogle.auth.creds import ServiceAccountCreds
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.utils import TIKA_SUPPORTED_FILETYPES, convert_to_b64, get_pem_format
+
+DEFAULT_RETRY_COUNT = 3
+DEFAULT_WAIT_MULTIPLIER = 2
+DEFAULT_FILE_SIZE_LIMIT = 10485760
+
+GOOGLE_DRIVE_READ_ONLY_SCOPE = "https://www.googleapis.com/auth/drive.readonly"
+API_NAME = "drive"
+API_VERSION = "v3"
```

Comment:
Idk if it's worth to put to constants - it seems to be used in a single place, so can either be moved to the only class using it or even inlined IMO